src/openrct2/peep/GuestPathfinding.cpp

```diff
@@ -855,17 +851,14 @@ static void peep_pathfind_heuristic_search(
                 }
                 else
                 { // numEdges == 2
-                    if (tileElement->AsPath()->IsQueue()
-                        && tileElement->AsPath()->GetRideIndex() != gPeepPathFindQueueRideIndex)
+
+                    bool rideQueueCheck = path->IsQueue() && peep.Is<Guest>()
+                        && path->GetRideIndex() != peep.As<Guest>()->GuestHeadingToRideId && !path->GetRideIndex().IsNull();
```

Comment:
This is not nice to read can you break it up into separate statements.

Reply:
Sure, I will do that while I am figuring out why changing the tileElement->AsPath() to a variable as Broxier suggested is causing the warning about potentially dereferencing a nullptr to act up in the Linux bullseye and jammy builds.

Reply:
Just do a nullptr check after the aspath